### ISSUE TITLE
All Typescript modules fail to run due to `Transforming JavaScript decorators`

### ISSUE THREAD

**Issue by sipsma** [dagger/dagger#7583]
(related PR https://github.com/dagger/dagger/pull/7573, creating an issue for tracking+visibility)

Currently, all typescript modules will fail to run due to a breaking change pushed to `tsx` ([issue](https://github.com/privatenumber/tsx/issues/582)), which the typescript SDK did not pin the version of in all previous engine releases.
* If your engine has a local cache of previous typescript SDK calls, it may work for a while until that local cache gets pruned

The error message looks like:
```
/src/dagger/src/index.ts:3:0: ERROR: Transforming JavaScript decorators to the configured target environment ("node21.3.0") is not supported yet
```

---

## Workaround until next release
This will be fixed in our next release (planned for early in the week of June 17-21), but if you are blocked by this in the meantime, a temporary workaround is to use a wrapper of the typescript SDK with the fix.

To initialize a new typescript module with this SDK, run
```
dagger init --sdk github.com/dagger/dagger/sdk/typescript/standalone@2cbda9c7547809a35eb93ff7677e5bab4afd00c1 <rest of dagger init args>
```

To change an existing module to use that SDK, edit `dagger.json` to specify
```
"sdk": "github.com/dagger/dagger/sdk/typescript/standalone@2cbda9c7547809a35eb93ff7677e5bab4afd00c1"
```

After our next release with the fix for this we'll close this issue and you can change back to using `typescript` as the SDK.

We also will improve this in the future so the ability to version SDKs separate from the engine is easier, initial proposal here: https://github.com/dagger/dagger/issues/7585

**Comment by harleylang** [comment]
Following the upstream tsx fix in https://github.com/privatenumber/tsx/issues/582 I retested a pipeline that was working prior to the tsx upstream breakage, and this is the new error observed:

```
Stderr:
node:internal/modules/cjs/loader:1181
  const err = new Error(`Cannot find module '${request}'`);
              ^
Error: Cannot find module '/src/pipelines/node/node_modules/@dagger.io/dagger/dist/index.js'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1181:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1169:15)
    at resolveExports (node:internal/modules/cjs/loader:591:14)
    at Module._findPath (node:internal/modules/cjs/loader:668:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1130:27)
    at resolve (/usr/local/lib/node_modules/tsx/dist/register-DIqrWwRV.cjs:1:2193)
    at /usr/local/lib/node_modules/tsx/dist/register-DIqrWwRV.cjs:1:2776
    at m._resolveFilename (file:///usr/local/lib/node_modules/tsx/dist/register-AuYqD7n5.mjs:1:831)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19) {
  code: 'MODULE_NOT_FOUND',
  path: '/src/pipelines/node/node_modules/@dagger.io/dagger/package.json'
}
Node.js v21.3.0
Error: input: module.withSource.initialize resolve: failed to initialize module: failed to call module "node" to get functions: call constructor: process "tsx --no-deprecation --tsconfig /src/pipelines/node/tsconfig.json /src/pipelines/node/src/__dagger.entrypoint.ts" did not complete successfully: exit code: 1
Stderr:
node:internal/modules/cjs/loader:1181
  const err = new Error(`Cannot find module '${request}'`);
              ^
Error: Cannot find module '/src/pipelines/node/node_modules/@dagger.io/dagger/dist/index.js'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1181:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1169:15)
    at resolveExports (node:internal/modules/cjs/loader:591:14)
    at Module._findPath (node:internal/modules/cjs/loader:668:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1130:27)
    at resolve (/usr/local/lib/node_modules/tsx/dist/register-DIqrWwRV.cjs:1:2193)
    at /usr/local/lib/node_modules/tsx/dist/register-DIqrWwRV.cjs:1:2776
    at m._resolveFilename (file:///usr/local/lib/node_modules/tsx/dist/register-AuYqD7n5.mjs:1:831)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19) {
  code: 'MODULE_NOT_FOUND',
  path: '/src/pipelines/node/node_modules/@dagger.io/dagger/package.json'
}
Node.js v21.3.0
```

**Comment by TomChv** [comment]
I can also confirm the issue has been fixed by the latest version of  `tsx` `v4.13.3`!

@sipsma I let you close this issue if you're okay with that
